keycloak-apikey/src/main/java/eu/europeana/keycloak/validation/provider/ApiKeyValidationProvider.java

```diff
@@ -30,21 +41,52 @@ public void close() {
     //specific implementation not required . e.g resource or connection cleanup
   }
 
-  @Path("")
+  @Path("/validate")
   @POST
   public Response validateApiKey(@QueryParam("client_id") String clientId , @QueryParam("ip") String ip ) {
     //validate token,apikey then IP in that order
-    ValidationResult result = service.validateAuthToken();
-    if (result == null) {
+    ValidationResult result = service.validateAuthToken(Constants.GRANT_TYPE_CLIENT_CRED);
+    if (result.getErrorResponse() == null) {
       result = service.validateApikey(clientId);
     }
-    if(result == null){
+    if(result.getErrorResponse() == null){
       result = service.validateIp(ip);
     }
     //TODO - Update Logic to consume the validated IP
-    if(result != null) {
+    if(result.getErrorResponse() != null) {
+      return Response.status(result.getHttpStatus()).entity(result.getErrorResponse()).build();
+    }
+    return Response.status(Status.NO_CONTENT).build();
+  }
+
+
+  @Path("/{client_public_id}")
+  @DELETE
+  public Response disableApikey(@PathParam("client_public_id") String client_public_Id){
+
+    ValidationResult result = service.validateAuthToken(Constants.GRANT_TYPE_PASSWORD);
+    if (result.getErrorResponse()==null) {
+      result = service.validateClientById(client_public_Id);
+    }
+    if (result.getErrorResponse() != null) {
       return Response.status(result.getHttpStatus()).entity(result.getErrorResponse()).build();
     }
+    if(!disableKey(client_public_Id, result.getUser())){
```

Comment:
Usually what I prefer is to have a if without `!` . 
It is just easier to analyse the condition as the name itself says: disablekey if true, just add that block there and the else part to be retuned by default

But it's just a preference of coding